File: tests/ruby/iso_b.rb
# iso_b.rb — see iso_a.rb: same hook names, own state, both must run.

def init
  @e = Labelle::Entity.create
  @e.set("IsoB", n: 0)
end

def update(dt)
  c = @e.get("IsoB")
  c[:n] += 10
  @e.set("IsoB", c)
end

def deinit
  Labelle.log("iso_b deinit")
end

File: tests/ts/iso_b.js
// iso_b.js — the other half of the module-isolation pair (see iso_a.js):
// identical binding and hook names, different state and step size.

let e = null;
let n = 0;

export function update(dt) {
  if (e === null) e = Entity.create();
  n += 10;
  e.set("IsoB", { n });
}

export function deinit() {
  labelle.log("iso_b deinit");
}
